### PR TITLE
MBS-11940: Properly error for invalid XML sent to /ws/2/tag

### PR DESCRIPTION
**Pull request by mwiencek**
https://tickets.metabrainz.org/browse/MBS-11940

### CODE REVIEW

File: lib/MusicBrainz/Server/Controller/WS/2/Tag.pm [hunk 1]
```diff
@@ -66,9 +66,16 @@ sub tag_submit : Private
     $self->_validate_post($c);
 
     my $xp = MusicBrainz::Server::WebService::XML::XPath->new( xml => $c->request->body );
+    my @nodelist;
+    eval {
+        @nodelist = $xp->find('/mb:metadata/*/*')->get_nodelist;
+    };
+    if ($@) {
```
Comment (reosarevok) [inline lib/MusicBrainz/Server/Controller/WS/2/Tag.pm:73]: I'm a bit confused, what does this mean here? :)

Comment (mwiencek) [reply lib/MusicBrainz/Server/Controller/WS/2/Tag.pm:73]: Just one of those special perl vars you have to memorize.  If there's an error inside the `eval`, Perl sets `$@` to the error: https://perldoc.perl.org/perlvar#%24%40

Comment (reosarevok) [reply lib/MusicBrainz/Server/Controller/WS/2/Tag.pm:73]: So it's effectively a try-catch, ok :)

**Review by reosarevok — APPROVED** [review]

**Review by yvanzo — APPROVED** [review]
Thanks

**Pull request merged**